### ISSUE TITLE
[feature request] Inconsistency of new Decoder api and Dense layer

### ISSUE THREAD

**Issue by persiyanov** [tensorflow/tensorflow#11332]
TF 1.1 has released new seq2seq API (`tf.contrib.seq2seq.BasicDecoder` etc.) and Decoders have `output_layer` parameter, which must be subclass of `tf.layers.Layer`.

If one wants to use dense projection, it has to do `from tensorflow.python.layers.core import Dense` which seems a bit inconvenient. 

Maybe the better way is to include `Dense` and `Dropout` classes into `tf.layers`? After that, one could write just `tf.layers.Dense` or `tf.layers.Dropout`.

**Comment by persiyanov** [comment]
This seems like a small fix in https://github.com/tensorflow/tensorflow/blob/r1.2/tensorflow/python/layers/layers.py, I can prepare it if it makes sense.

**Comment by cy89** [comment]
I checked internally, and it sounds like we are planning to address this in the next 2-4 weeks, after a few other internal dependencies are worked out (constraints API and Network class). Thanks for offering to make the fix; I think the right thing is to close for now, and ask you to reopen if you don't see an appropriate change roll out in a month or so.

**Comment by persiyanov** [comment]
Ok, thanks for the detailed response:)
